src/main/java/Duke.java

```diff
@@ -1,10 +1,204 @@
+            taskList.add(t);
+            UI.printTask(taskList);
+        } catch (EmptyToDoDescriptionException e){
+            UI.printEmptyToDoDescriptionException();
+        }
+    }
+
+    static void event(String line) {
+        try {
+            checkEmptyEventDescription(line.trim());
+            //e.g event dinner /at 12 oct 2019
+            if (line.contains("/at")) {
+                //str[1] to get date - 12 oct 2019
+                String[] str = line.split(" /at ", 2);
+                //description[1] to get description - dinner
+                String[] taskDescription = str[0].split(" ", 2);
+                SimpleDateFormat dateFormat = new SimpleDateFormat("dd MMM yyyy");
+                //Format date 12 Oct 2019, to string "12 Oct 2019"
+                Date date = dateFormat.parse(str[1]);
+                String strDate = dateFormat.format(date);
+                Task t = new Event(taskDescription[1], strDate);
+                taskList.add(t);
+            } else {
+                //e.g event dinner
+                String taskDescription = line.substring(6);
+                Task t = new Event(taskDescription, "Date not specified");
+                taskList.add(t);
+            }
+            UI.printTask(taskList);
+        } catch (EmptyEventDescriptionException e) {
+            UI.printEmptyEventDescriptionException();
+        } catch (ParseException e) {
+            UI.printParseException();
+        }
+    }
+
+    static void deadline(String line) {
+        try {
+            checkEmptyDeadlineDescription(line.trim());
+            if (line.contains("/by")) {
+                //str[1] to get date - 12 oct 2019
+                String[] str = line.split(" /by ", 2);
+                //description[1] to get description - duke
+                String[] taskDescription = str[0].split(" ", 2);
+                SimpleDateFormat dateFormat = new SimpleDateFormat("dd MMM yyyy");
+                //Format date 12 Oct 2019, to string "12 Oct 2019"
+                Date date = dateFormat.parse(str[1]);
+                String strDate = dateFormat.format(date);
+                Task t = new Deadline(taskDescription[1], strDate);
+                taskList.add(t);
+            } else {
+                //e.g deadline duke
+                String taskDescription = line.substring(9);
+                Task t = new Deadline(taskDescription, "Date not specified");
+                taskList.add(t);
+            }
+            UI.printTask(taskList);
+        } catch (EmptyDeadlineDescriptionException e) {
+            UI.printEmptyDeadlineDescriptionException();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    static void list() {
+        try {
+            listEmpty(taskList);
+            UI.printOutput((taskList));
+        } catch (ListEmptyException e) {
+            UI.printListEmpty();
+        }
+    }
+
+    static void done(String line) {
+        try {
```

Comment:
The done method here seems convoluted a bit? Maybe some comments as to what it does with the user input line would be very helpful?

Reply:
Thanks for the advice, will improvise the code accordingly!